packages/taiko-client/driver/preconf_blocks/server.go

```diff
@@ -177,6 +250,236 @@ func (s *PreconfBlockAPIServer) SetP2PSigner(p2pSigner p2p.Signer) {
 	s.p2pSigner = p2pSigner
 }
 
+// preconfTransitionActive returns true if the current time has reached or passed
+// the configured transition timestamp, and the timestamp is non-zero.
+func (s *PreconfBlockAPIServer) preconfTransitionActive() bool {
+	if s.transitionTimestamp == 0 {
+		return false
+	}
+	now := uint64(time.Now().Unix())
+	return now >= s.transitionTimestamp
+}
+
+// OnUnsafePreconfirmationCommitment handles preconfirmation commitments gossiped over P2P.
+// It validates the slasher address and defers block reconstruction to the driver/engine.
+func (s *PreconfBlockAPIServer) OnUnsafePreconfirmationCommitment(
+	ctx context.Context,
+	from peer.ID,
+	sc *p2p.SignedCommitment,
+) error {
+	if !s.preconfTransitionActive() {
+		log.Debug("Ignoring preconfirmation commitment before transition", "peer", from)
+		return nil
+	}
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	start := time.Now()
+	defer func() {
+		elapsedMs := time.Since(start).Milliseconds()
+		log.Debug("OnUnsafePreconfirmationCommitment completed", "elapsed", fmt.Sprintf("%dms", elapsedMs))
+	}()
+
+	// Ignore our own messages
+	if s.p2pNode != nil && from == s.p2pNode.Host().ID() {
+		log.Debug("Ignore preconfirmation from self", "peer", from)
+		return nil
+	}
+
+	if sc == nil {
+		log.Warn("Empty preconfirmation commitment received", "peer", from)
+		return nil
+	}
+
+	// Validate slasher address matches expected committer for current window
+	expected := s.CurrentPreconferCommitter()
```

Comment:
this is incorrect, the slasher address is the address *of the actual slasher contract*, we only need to verify the SlasherAddress on the commitment is equal to the actual slasher address (which we can pass in to the driver via a flag)